calico_node/filesystem/startup.py

```diff
@@ -261,7 +261,7 @@ def main():
     # This is required because the confd templates expect to be able to fill in
     #  the IP by fetching it from the environment.
     with open('startup.env', 'w') as f:
-        f.write("IP=" + ip)
+        f.write("IP=" + str(ip))
```

Comment:
I think this should be done at line 250.  All of the calls below should be expecting the string version of the `ip`. The calls below expect string IPs as well, but again, we don't have strong testing for this in STs.